orderapi/src/main/java/fi/hel/verkkokauppa/order/service/order/OrderService.java

```diff
@@ -433,4 +453,69 @@ public String getFirstMerchantId( OrderAggregateDto order){
                 .findFirst()
                 .orElse(null);
     }
+
+    public String collectDetailedInformationForNotification(String orderId) {
+        try {
+            // Get extra text message from service for detailed order information
+
+            // collect order and merchant data
+
+            // get Order
+            OrderAggregateDto orderDto = orderAggregateDto(orderId).getBody();
+            String namespace = orderDto.getOrder().getNamespace();
+
+            // Get Payment
+            JSONObject paymentResponse = restServiceClient.makeAdminGetCall(serviceUrls.getPaymentServiceUrl() + "/payment-admin/online/get?orderId=" + orderId);
+            ObjectMapper objectMapper = new ObjectMapper();
+            PaymentDto paymentDto = objectMapper.readValue(paymentResponse.toString(), PaymentDto.class);
```

Comment:
Move this get payment to function and add try catch to log error and return null, do not throw error